core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala

```diff
@@ -170,7 +171,16 @@ trait EndToEndAuthorizationTest extends IntegrationTestHarness with SaslSetup {
     * Tests the ability of producing and consuming with the appropriate ACLs set.
     */
   @Test
-  def testProduceConsume {
+  def testProduceConsumeViaAssign {
+    testProduceConsume(assign = true)
+  }
+
+  @Test
+  def testProduceConsumeViaSubscribe {
+    testProduceConsume(assign = false)
+  }
+
+  private def testProduceConsume(assign : Boolean) {
```

Comment:
It's certainly subjective. Personally, `private` methods should follow similar rules as non-private unless there is a good reason not to. A boolean named `assign` in a method called `testProduceConsume` is non-obvious and it seems like the alternative I suggested would be clearer. In my opinion, of course.